### PR TITLE
fix: e2e test failed

### PR DESCRIPTION
**Pull request by lei9444**
## Description
image
The data collection dialog should be closed when run e2e test

## Task Item
#minor

### CODE REVIEW

File: Composer/cypress/support/commands.ts [hunk 1]
```diff
@@ -4,6 +4,12 @@
 import '@testing-library/cypress/add-commands';
 
 Cypress.Commands.add('createBot', (botId: string, botName?: string) => {
+  cy.get('body').then((body) => {
+    if (body.find('[data-testid="data collection dialog"]').length > 0) {
+      cy.findByText('Not now').click();
+    }
+  });
```
Comment (a-b-r-o-w-n) [inline Composer/cypress/support/commands.ts:11]: nit:

```ts
if (cy.findByTestId('data collection dialog')) {
  // ...
}
```

File: Composer/packages/client/src/components/DataCollectionDialog.tsx [hunk 2]
```diff
@@ -23,6 +23,7 @@ const DataCollectionDialog: React.FC = () => {
 
   return (
     <Dialog
+      data-testid="data collection dialog"
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/components/DataCollectionDialog.tsx:26]: nit: camel case this.

**Review by a-b-r-o-w-n — CHANGES_REQUESTED** [review]
This is a fine thing to do, but here are a couple of questions:

1. Can we just disable this altogether when not in production?
2. If not, is there an api we can interact with during set up that allows us to bypass interacting with the ui?
3. I don't think this is the right place to put this code since it probably only happens on a new user session. It doesn't have anything to do with creating a bot. It should go in a set up function.

@tdurnford do you have any thoughts about the first 2 questions?

**Comment by hatpick** [comment]
I'm with @a-b-r-o-w-n on disabling it when not in production.

**Comment by tdurnford** [comment]
@sorgh @a-b-r-o-w-n I think disabling it when it's not in production is probably the best approach.

**Comment by a-b-r-o-w-n** [comment]
@tdurnford Something to consider is if we run the e2e tests with a prod build or not. I don't recall.

**Comment by tdurnford** [comment]
@a-b-r-o-w-n We run it with a production build. I'm trying to set `allowDataCollection` to false in the `Composer/cypress/__e2e_data.json`, but I'm running into issues with that at the moment.

**Comment by lei9444** [comment]
this will be fixed in #4907 .
close this

**Pull request closed without merging**